src/fuzz/dummy_fuzz_test.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <iostream>
+#include <stdexcept>
+#include <filesystem>
+
+#include "frontend/server/server.h"
+#include "google/cloud/spanner/client.h"
+#include "google/cloud/spanner/create_instance_request_builder.h"
+#include "google/cloud/spanner/database_admin_client.h"
+#include "google/cloud/spanner/instance_admin_client.h"
+
+using Server = ::google::spanner::emulator::frontend::Server;
+using ::google::cloud::future;
+using ::google::cloud::StatusOr;
+using ::google::cloud::spanner::DatabaseAdminClient;
+using ::google::cloud::spanner::v0::ConnectionOptions;
+
+#ifdef __OSS_FUZZ__
+#define OVERWRITE 1
+
+bool DoOssFuzzInit() {
+  namespace fs = std::filesystem;
+  fs::path originDir;
+  try {
+    originDir = fs::canonical(fs::read_symlink("/proc/self/exe")).parent_path();
+  } catch (const std::exception& e) {
+    return false;
+  }
+  fs::path tzdataDir = originDir / "data/zoneinfo/";
+  if (setenv("TZDIR", tzdataDir.c_str(), OVERWRITE)) {
+    return false;
+  }
+  return true;
+}
+#endif
+
+const std::string server_address = "localhost:1234";
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+  #ifdef __OSS_FUZZ__
+    static bool Initialized = DoOssFuzzInit();
+    if (!Initialized) { std::abort(); }
+  #endif
+
+  Server::Options options;
+  options.server_address = server_address;
+  std::unique_ptr<Server> server = Server::Create(options);
```

Comment:
If you have designated initializers (C++20), they are great for initializing options structs:

std::unique_ptr<Server> server = Server::Create({.server_address = server_address});

If Server::Create is explicit (happens more often with constructors) then you may have to include the struct type:

std::unique_ptr<Server> server = Server::Create(Server::Options{.server_address = server_address});

Reply:
My comment was deleted but the project is using C++17 so will hold off on this for now